### PR TITLE
Ledger update

### PR DESCRIPTION
**Pull request by bucko13**
cleanup and consolidation of two other outstanding PRs #11 and #9

### CODE REVIEW

File: src/ledger.js [hunk 1]
```diff
@@ -896,3 +990,24 @@ export class LedgerSignMultisigTransaction extends LedgerBitcoinInteraction {
...
 }
+
+/** Local helper functions **/
+
+/**
+ * get the path of the parent of the given path
+ * @param {string} bip32Path 
+ * @returns {string}
+ */
+function getParentPath(bip32Path) {
```
Comment (bucko13) [inline src/ledger.js]: going to move this into unchained wallets

File: src/ledger.js [hunk 2]
```diff
@@ -627,7 +634,94 @@ export class LedgerExportPublicKey extends LedgerExportHDNode {
...
+   * Get fingerprint for given path. This is useful for generating xpubs
+   * which need the fingerprint of the parent pubkey
+   * @param {string} path - path of pubkey to retrieve fingerprint for
+   * @returns {string} fingerprint
+   */
+   async getFingerprint(path) {
+    let fingerprint
+    await this.withApp(async (app) => {
+      const key = await app.getWalletPublicKey(path)
+      let compressed = compressPublicKey(key.publicKey)
```
Comment (bucko13) [inline src/ledger.js]: I'd like to move this into an `unchained-bitcoin` utility too.

File: src/ledger.js [hunk 3]
```diff
@@ -627,7 +634,94 @@ export class LedgerExportPublicKey extends LedgerExportHDNode {
...
+   * @returns {string} extended public key(xpub) for the BIP32 path of a given instance
+   */
+  async run() {
+    const walletPublicKey = await super.run();
+    const compressedPubkey = compressPublicKey(walletPublicKey.publicKey)
+
+    let node;
+    await this.withApp(async (app) => {
+      // get the hd wallet node and fill in missing properties
+      node = bip32.fromPublicKey(Buffer.from(compressedPubkey, 'hex'),
```
Comment (bucko13) [inline src/ledger.js]: this probably could be pulled out into a `getHDWalletNode` utility function in `unchained-bitcoin` too.

**Comment by bucko13** [comment]
The build is failing because the unchained-bitcoin dependency is set to the local branch right now. I'll update this after https://github.com/unchained-capital/unchained-bitcoin/pull/7 is approved and published.

File: package.json [hunk 4]
```diff
@@ -49,7 +50,7 @@
     "pako": "^1.0.10",
     "punycode": "^2.1.1",
     "trezor-connect": "^8.0.13",
-    "unchained-bitcoin": "0.0.7"
+    "unchained-bitcoin": "file:../unchained-bitcoin"
```
Comment (dhruvbansal) [inline package.json]: Need to switch this to the correct `unchained-bitcoin` version.

Comment (bucko13) [reply package.json]: Done!

**Pull request merged**